stl/inc/algorithm

```diff
@@ -7405,6 +7608,339 @@ void inplace_merge(_ExPo&&, _BidIt _First, _BidIt _Mid, _BidIt _Last) noexcept /
+                *_First = _RANGES iter_move(_Mid); // the lowest element is now in position
+                ++_First;
+                ++_Mid;
+                if (_Mid == _Last) {
+                    // move the remaining left partition
+                    _RANGES _Move_unchecked(_Left_first, _Backout._Last, _First);
+                    return;
+                }
+            } else {
+                *_First = _RANGES iter_move(_Left_first);
+                ++_First;
+                ++_Left_first;
+                if (_Left_first == _Left_last) {
+                    // move the remaining right partition and highest element, since *_Left_first is highest
+                    const auto _Final = _RANGES _Move_unchecked(_Mid, _Last, _First);
+                    *_Final.out       = _RANGES iter_move(_Left_first);
+                    return;
+                }
+            }
+        }
+    }
+
+    // clang-format off
+    template <bidirectional_iterator _It, class _Pr, class _Pj>
+        requires sortable<_It, _Pr, _Pj>
+    void _Inplace_merge_buffer_right(_It _First, _It _Mid, _It _Last, iter_value_t<_It>* _Right_first,
+        const ptrdiff_t _Capacity, _Pr _Pred, _Pj _Proj) {
+        // clang-format on
+        // move the range [_Mid, _Last) to _Right_first, and merge it with [_First, _Mid) to _Last
+        // usual invariants apply
+        using _Ty = iter_value_t<_It>;
+
+        _Ty* _Right_last =
+            _RANGES _Uninitialized_move_unchecked(_Mid, _Last, _Right_first, _Right_first + _Capacity).out;
+        _Uninitialized_backout<_Ty*> _Backout(_Right_first, _Right_last);
+
+        // we already know, that _Mid points to the next highest element and that there is more than 1 element left.
+        *--_Last = _RANGES iter_move(--_Mid);
+
+        // we already know, that _Backout._Last - 1 is the highest element so do not compare against it again.
+        --_Mid;
+        --_Right_last;
+        for (;;) {
+            if (_STD invoke(_Pred, _STD invoke(_Proj, *_Right_last), _STD invoke(_Proj, *_Mid))) {
+                *--_Last = _RANGES iter_move(_Mid); // the lowest element is now in position
+                if (_First == _Mid) {
+                    *--_Last = _RANGES iter_move(_Right_last); // to make [_Right_first, _Right_last) a half-open range
+                    _RANGES _Move_backward_common(_Right_first, _Right_last, _STD move(_Last));
+                    return;
+                }
+                --_Mid;
+            } else {
+                *--_Last = _RANGES iter_move(_Right_last);
+                --_Right_last;
+                if (_Right_first == _Right_last) { // we can't compare with *_Right_first, but we know it is lowest
+                    *--_Last = _RANGES iter_move(_Mid); // restore half-open range [_First, _Mid)
+                    _RANGES _Move_backward_common(_First, _STD move(_Mid), _STD move(_Last));
+                    *_First = _RANGES iter_move(_Right_first);

```

Comment:
That one is strange. I belive we acutally move to [_First, _Mid) in the line above?